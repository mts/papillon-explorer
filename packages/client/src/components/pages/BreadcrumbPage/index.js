import React from 'react';
import { Link } from 'react-router-dom';

const BreadcrumbPage = () => {
  return (
    <div>
      <h2>Breadcrumb Page</h2>
      <div>
        Under construction ğŸ™‚
        <Link to="/">Back to Home Page</Link>
      </div>
    </div>
  )
}

export default BreadcrumbPage;
