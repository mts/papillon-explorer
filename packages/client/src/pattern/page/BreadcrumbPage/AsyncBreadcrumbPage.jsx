import React from 'react'

const LazyBreadcrumbPage = React.lazy(() => import('./BreadcrumbPage'))

export const AsyncBreadcrumbPage = () => (
  <React.Suspense fallback={null}>
    <LazyBreadcrumbPage />
  </React.Suspense>
)
