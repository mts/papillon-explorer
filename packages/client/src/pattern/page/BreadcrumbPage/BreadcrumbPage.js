import React from 'react'
import { Query } from 'react-apollo'
import { startPageRoute } from '../../../routes'
import { PAGES } from '../../../graphql/query'
import BreadcrumbTemplate from '../../template/BreadcrumbTemplate'

const BreadcrumbPage = () => {
  return (
    <Query query={PAGES}>
      {({ data }) =>
        data.pages && data.pages.pages ? (
          <BreadcrumbTemplate page={data.pages.pages.find(page => page.id === 'breadcrumb')} startPageRoute={startPageRoute} />
        ) : null
      }
    </Query>
  )
}

export default BreadcrumbPage
