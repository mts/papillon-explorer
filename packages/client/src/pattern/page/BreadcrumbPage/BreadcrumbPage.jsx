import React from 'react'
import { useQuery } from '@apollo/react-hooks'
import { PAGES } from '../../../graphql/query'
import { BreadcrumbTemplate } from '../../template/BreadcrumbTemplate'

export const BreadcrumbPage = () => {
  const { data } = useQuery(PAGES)

  if (data && data.pages && data.pages.pages) {
    return <BreadcrumbTemplate page={data.pages.pages.find(page => page.id === 'breadcrumb')} />
  }

  return null
}

export default BreadcrumbPage
