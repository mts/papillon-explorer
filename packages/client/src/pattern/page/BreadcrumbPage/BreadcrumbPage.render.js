import React from 'react'
import { BreadcrumbPage } from './BreadcrumbPage'

export const breadcrumbPageDefault = <BreadcrumbPage />
