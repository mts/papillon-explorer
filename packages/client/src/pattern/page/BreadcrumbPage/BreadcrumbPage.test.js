import React from 'react'
import { breadcrumbPageDefault } from './BreadcrumbPage.render'

jest.mock('@apollo/react-hooks', () => {
  return {
    useQuery: () => {
      return {
        data: {
          pages: {
            pages: [{ id: 'breadcrumb', name: 'some-name' }],
          },
        },
      }
    },
  }
})

jest.mock('react-router-dom', () => ({ Link: () => <div>some link</div> }))

describe('<BreadcrumbPage />', () => {
  describe('Snaphot', () => {
    test('must match breadcrumbPageDefault', () => {
      expect(global.renderToJSON(breadcrumbPageDefault)).toMatchSnapshot()
    })
  })
})
