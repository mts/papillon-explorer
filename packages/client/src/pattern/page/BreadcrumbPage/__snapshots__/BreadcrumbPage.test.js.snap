// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadcrumbPage /> Snaphot must match breadcrumbPageDefault 1`] = `
<div>
  <h2>
    some-name
  </h2>
  <div>
    Under construction
     
    <span
      aria-label="emoji"
      role="img"
    >
      ðŸ™‚
    </span>
    <div>
      some link
    </div>
  </div>
</div>
`;
